### DISCUSSION TITLE
For iOS, the default Accessibility ID is set to the name of the UI element.

### DISCUSSION THREAD

**Discussion by chetnaag95** [appium/appium#18669] [Q&A]
Hi I am adding accessibility-label in my react native app. I am getting accessibility id as UI element name in iOS simulator but for Android, the value of Accessibility is same as the value of the attribute “content-desc”.

Is there any way I can get same value of accessibility id in iOS as well as I am getting in android so that it will be easier at the time of automation

**Comment by KazuCocoa** [comment]
You could find the `label` with `predicate` for iOS
https://appium.github.io/appium-xcuitest-driver/4.29/ios-predicate/

**Comment by chetnaag95** [comment]
@KazuCocoa I am setting same accessibility label for ios and android in react native but I am getting different accessibility id in appium inspector. For iOS it is set to name of the element and for android it is set as content-desc

**Reply by KazuCocoa** [reply]
Maybe the ReactNative sets another value for the element's `accessibilityIdentifier` (https://developer.apple.com/documentation/uikit/uiaccessibilityidentification)

Appium/WDA returns the identifier first, then label as https://github.com/appium/WebDriverAgent/blob/0cb36e656126755196efc243af66468c4d115c92/WebDriverAgentLib/Categories/XCUIElement%2BFBWebDriverAttributes.m#L107-L112 for it.

https://github.com/appium/WebDriverAgent/blob/master/WebDriverAgentLib/Commands/FBFindElementCommands.m#L175-L179
https://github.com/appium/WebDriverAgent/blob/0cb36e656126755196efc243af66468c4d115c92/WebDriverAgentLib/Categories/XCUIElement%2BFBFind.m#LL125C70-L125C85
